src/sentry/sentry_metrics/querying/visitors/query_expression.py

```diff
@@ -271,10 +277,27 @@ def _group_bys_as_string(self, group_bys: list[Column | AliasedExpression] | Non
 
         string_group_bys = set()
         for group_by in group_bys:
+            selected_mapper = None
+
             if isinstance(group_by, AliasedExpression):
-                string_group_bys.add(group_by.exp.name)
+                for mapper in self.mappers:
+                    if mapper.to_key == group_by.exp.name:
+                        selected_mapper = mapper
+                        break
             elif isinstance(group_by, Column):
-                string_group_bys.add(group_by.name)
+                for mapper in self.mappers:
+                    if mapper.to_key == group_by.name:
+                        selected_mapper = mapper
+                        break
```

Comment:
nit: lift this into its own function that returns the matching mapper.

Reply:
Changed